### PR TITLE
Bug 1173087 - Add YouTube search button

### PR DESCRIPTION
**Pull request by mystic123**
Added youtube search engine which provides youtube search and search suggestions

### CODE REVIEW

**Comment by AaronMT** [comment]
Thanks for the PR. Does this need product team decision?

**Comment by mystic123** [comment]
Well. I don’t know. This is my first pull request ever.

**Comment by AaronMT** [comment]
Sorry that was a question for the mobile team at Mozilla.

**Comment by fluffyemily** [comment]
This is good but it only works with US English and no other locales (testing it in UK English first time didn't make You Tube appear, I had to change region). We need to make a decision as to which locales we would want YouTube to appear for and migrate these changes to all necessary.

It raises an issue with our search term detector as when You Tube redirects from desktop to mobile, it also opens a new page (so desktop -> mobile -> new page with results), which means our method of reserve engineering the search URL to get the terms doesn't work as the initial URL is further back than the current `WKBackForwardItem`. A bug should be raised for that when this lands.

**Comment by juancasas** [comment]
Bien

File: Client/Assets/SearchPlugins/en/list.txt [hunk 1]
```diff
@@ -4,4 +4,5 @@ twitter
 wikipedia
 yahoo
 bing
-duckduckgo
\ No newline at end of file
+duckduckgo
```
Comment (aterribili) [inline Client/Assets/SearchPlugins/en/list.txt]: :clap:

**Comment by st3fan** [comment]
Hi @mystic123, thank you for your patch. We have decided to not land this because are including a feature that will let you add any site to the search engines. You can follow this feature at https://bugzilla.mozilla.org/show_bug.cgi?id=1205877

**Pull request closed without merging**